src/Game/zCombo.cpp

```diff
@@ -2,23 +2,208 @@
+    }
+
+    for (int i = 0; i < 16; ++i)
+    {
+        id = xStrHash(comboReward[i].textName);
+        comboReward[i].textAsset = (xTextAsset*)xSTFindAsset(id, 0);
+    }
+
+    comboReward[0].reward = globals.player.g.ShinyValueCombo0;
+    comboReward[1].reward = globals.player.g.ShinyValueCombo1;
+    comboReward[2].reward = globals.player.g.ShinyValueCombo2;
+    comboReward[3].reward = globals.player.g.ShinyValueCombo3;
+    comboReward[4].reward = globals.player.g.ShinyValueCombo4;
+    comboReward[5].reward = globals.player.g.ShinyValueCombo5;
+    comboReward[6].reward = globals.player.g.ShinyValueCombo6;
+    comboReward[6].reward = globals.player.g.ShinyValueCombo6;
+    comboReward[8].reward = globals.player.g.ShinyValueCombo8;
+    comboReward[9].reward = globals.player.g.ShinyValueCombo9;
+    comboReward[10].reward = globals.player.g.ShinyValueCombo10;
+    comboReward[11].reward = globals.player.g.ShinyValueCombo11;
+    comboReward[12].reward = globals.player.g.ShinyValueCombo12;
+    comboReward[13].reward = globals.player.g.ShinyValueCombo13;
+    comboReward[14].reward = globals.player.g.ShinyValueCombo14;
+    comboReward[15].reward = globals.player.g.ShinyValueCombo15;
+    comboMaxTime = globals.player.g.ComboTimer;
+
+    for (int i = 0; i < 16; ++i)
+    {
+        fillCombo(&comboReward[i]);
+    }
+
+    // "TEXTBOX_BUNGEE_HELP"
+    id = xStrHash(zCombo_Strings + 0xd7);
+    sHideText[0] = zSceneFindObject(id);
+
+    // "DIALOG_TEXTBOX"
+    id = xStrHash(zCombo_Strings + 0xeb);
+    sHideText[1] = zSceneFindObject(id);
+
+    // "MESSAGE_02_TEXTBOX"
+    id = xStrHash(zCombo_Strings + 0xfa);
+    sHideText[2] = zSceneFindObject(id);
+
+    // "PROMPT_TEXTBOX"
+    id = xStrHash(zCombo_Strings + 0x10d);
+    sHideText[3] = zSceneFindObject(id);
+
+    // "QUIT_TEXTBOX"
+    id = xStrHash(zCombo_Strings + 0x11c);
+    sHideText[4] = zSceneFindObject(id);
+
+    // "MNU4 NPCTALK"
+    id = xStrHash(zCombo_Strings + 0x129);
+    sHideUIF[0] = zSceneFindObject(id);
+}
+#endif
 
 // func_80195C10
+#if 1
 #pragma GLOBAL_ASM("asm/Game/zCombo.s", "zCombo_Add__Fi")
+#else
+/* Can't figure out how to get the assignments to happen in the right order */
+void zCombo_Add(int32 points)
+{
+    if (comboTimer < lbl_803CFBE4) {
+        comboTimer = comboMaxTime;
+        comboPending = points - 1;
+    } else {
+        comboTimer = comboMaxTime;
+        comboCounter += points;
+        if (comboPending != 0) {
+            comboCounter += comboPending;
+            comboPending = 0;
+        }
+    }
+}
+#endif
 
 // func_80195C6C
-#pragma GLOBAL_ASM("asm/Game/zCombo.s", "zComboHideMessage__FRQ24xhud6widgetRQ24xhud6motive")
+void zComboHideMessage(xhud::widget& w, xhud::motive& motive)
+{
+    w.hide();
+}
 
 // func_80195C8C
-#pragma GLOBAL_ASM("asm/Game/zCombo.s", "zCombo_HideImmediately__Fv")
+void zCombo_HideImmediately()
+{
+    if (comboHUD != NULL) {
+        // This is probably wrong, but generates the right code
```

Comment:
I would add a `// TODO: blah blah blah` comment here, so anyone who searches for TODO can find it.